htsget_server/database.py

```diff
@@ -1,87 +1,189 @@
+            'checksums': json.loads(self.checksums),
+            'description': self.description,
+            'mime_type': self.mime_type,
+            'aliases': json.loads(self.aliases)
+        }
+        if len(list(self.contents)) > 0:
+            result['contents'] = json.loads(self.contents.__repr__())
+        if len(list(self.access_methods)) > 0:
+            result['access_methods'] = json.loads(self.access_methods.__repr__())
+        return json.dumps(result)
+
+
+class AccessMethod(ObjectDBBase):
+    __tablename__ = 'access_method'
+    id = Column(Integer, primary_key=True)
+    drs_object_id = Column(Integer, ForeignKey('drs_object.id'))
+    drs_object = relationship("DrsObject", back_populates="access_methods")
+    type = Column(String, default='')
+    access_id = Column(String, default='')
+    region = Column(String, default='')
+    url = Column(String, default='')
+    headers = Column(String, default='[]') # JSON array of strings
+
+    def __repr__(self):
+        result = {
+            'type': self.type
+        }
+        if self.region is not "":
+            result['region'] = self.region
+        if self.url is not "":
+            result['access_url'] = {
+                'url': self.url,
+                'headers': json.loads(self.headers)
+            }
+        if self.access_id is not "":
+            result['access_id'] = self.access_id
+
+        return json.dumps(result)
+
+
+class ContentsObject(ObjectDBBase):
+    __tablename__ = 'content_object'
+    id = Column(Integer, primary_key=True)
+    drs_object_id = Column(Integer, ForeignKey('drs_object.id'))
+    drs_object = relationship("DrsObject", back_populates="contents")
+    name = Column(String, default='') # like a filename
+    drs_uri = Column(String, default='[]') # JSON array of strings of DRS id URIs
+    contents = Column(String, default='[]') # JSON array of ContentsObject.ids
+    def __repr__(self):
+        result = {
+            'name': self.name,
+            'drs_uri': json.loads(self.drs_uri)
+        }
+        if len(json.loads(self.contents)) > 0:
+            result['contents'] = json.loads(self.contents)
+
+        return json.dumps(result)
+    
+    
+ObjectDBBase.metadata.create_all(engine)
+Session = sessionmaker(bind=engine)
+
+""" Helper Functions"""
+def get_drs_object(object_id, expand=False):
+    with Session() as session:
+        result = session.query(DrsObject).filter_by(id=object_id).one_or_none()
+        if result is not None:
+            new_obj = json.loads(str(result))
+#         if expand:
+#             expand doesn't do anything on this DRS server
+            return new_obj
+        return None
+
+
+def list_drs_objects():
+    with Session() as session:
+        result = session.query(DrsObject).all()
+        if result is not None:
+            new_obj = json.loads(str(result))
+            return new_obj
+        return None
+
+
+def create_drs_object(obj):
+    with Session() as session:
```

Comment:
Would that be superior? It seems like it would be basically the same, but would make this even longer, line-wise.

Reply:
But also in this case, I think you can reduce the code by using a an iterator for obj.keys() so you only set the object attributes the same way as you set dictionary keys/values

Reply:
Okay I tried to do this and it's basically impossible because of the SQLAlchemy declarative base thing, or at least it's not worth the effort. However, thank you for pushing me down a rabbit hole where I made some better integrity checks in the test suite.